### PR TITLE
Adding "CyberArk Conjur - Retrieve a Secret"

### PR DESCRIPTION
**Pull request by quincycheng**
### Step template checklist

- [X] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [X] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [X] Parameter names should not start with `$`
- [X] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [X] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [X] If a new `Category` has been created:
   - [X ] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [X] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

### CODE REVIEW

File: step-templates/cyberark-conjur-retrieve-a-secret.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      "DefaultValue": "True",
+      "DisplaySettings": {
+        "Octopus.ControlType": "Checkbox"
+      }
+    },
+    {
+      "Id": "91cfcddd-8f16-4aba-9f03-c8bb6dcf9218",
+      "Name": "FIX_SLASH_ENCODING",
+      "Label": "Fix Incorrect Slash Encoding",
+      "HelpText": "PowerShell may incorrectly decode slashes in URL.   If error 404 is returned, toggling this option may fix the issue",
```
Comment (jburger) [inline step-templates/cyberark-conjur-retrieve-a-secret.json:90]:
```suggestion
      "HelpText": "PowerShell may incorrectly decode slashes in URL. If an error 404 is returned, toggling this option may fix the issue",
```

**Review by jburger — CHANGES_REQUESTED** [review]
Thanks for the submission @quincycheng !

There are a couple of small things to tidy up here before I can approve the PR. 

Could you explain the need to encode slashes in the `VARIABLE_ID` parameter and `CreateUriWithoutIncorrectSlashEncoding` for us? I'm not familiar with CyberArk Conjur specifically, but it stands out as being an issue, given you've provided a parameter to optionally turn that off.

File: step-templates/cyberark-conjur-retrieve-a-secret.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      "Label": "Output Variable Name",
+      "HelpText": "This specifies the output variable.   For more details of output variables, please refer to https://octopus.com/docs/projects/variables/output-variables",
+      "DefaultValue": "Secret",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "28391ac5-8915-4c85-bc64-7d88a03b812c",
+      "Name": "STAY_SENSITIVE",
```
[the remark this answers is not in the document]
Comment (quincycheng) [reply step-templates/cyberark-conjur-retrieve-a-secret.json]: thanks.   updated.

File: step-templates/cyberark-conjur-retrieve-a-secret.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      "Label": "Variable ID of Conjur Secret",
+      "HelpText": " Variable ID of Conjur Secret",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Id": "738037a9-80ad-43a4-b6dd-6ff51330b84a",
+      "Name": "OUTPUT_NAME",
```
[the remark this answers is not in the document]
Comment (quincycheng) [reply step-templates/cyberark-conjur-retrieve-a-secret.json]: thanks.  updated.

File: step-templates/cyberark-conjur-retrieve-a-secret.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      "Label": "Conjur Authn API Key",
+      "HelpText": "User/host API key",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "Sensitive"
+      }
+    },
+    {
+      "Id": "ed1a91df-d0fa-453e-8b31-bdae379b593b",
+      "Name": "VARIABLE_ID",
```
[the remark this answers is not in the document]
Comment (quincycheng) [reply step-templates/cyberark-conjur-retrieve-a-secret.json]: thanks.  updated.

**Comment by quincycheng** [comment]
Thanks for reviewing, @jburger 

According to Conjur doc (see [link](https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Retrieve_Secret.htm)), the  identifier included in the URL must be URL-encoded.   However, due to some some PowerShell version & settings, those URL-encoded string in the URL could be automatically decoded by PowerShell, before the URL is sent to Conjur server.   And that causes the retrieval of secret fails.

I tested on OctopusDeploy SaaS and this happens, but the PowerShell on my local Windows 10  doesn't have such issue.

The reason why `CreateUriWithoutIncorrectSlashEncoding` function & a toggle checkbox are implemented to overcome this potential issue

**Comment by quincycheng** [comment]
I believe I have amended all the suggested changes.
Please let me know if I've missed any or any further changes are required
thanks

**Comment by DevOpsDerek** [comment]
Awaiting Internal approval from Security team

**Comment by quincycheng** [comment]
Noted that there are a few releases in past 2 weeks.
Anything that I can help to get this PR merged?    Thanks.

**Comment by harrisonmeister** [comment]
> I notice that there are a few releases in past 2 weeks.
> Anything that I can help to get this PR merged? Thanks.

Hi @quincycheng - apologies we will try to get to this in a day or so (our security team would like to rubber-stamp it :smile:)

**Review by jburger — APPROVED** [review]
Great! Thankyou so much @quincycheng and also for being patient with our process.

**Pull request merged**